### PR TITLE
Story1

### PR DESCRIPTION
**Pull request by hayfaa14**
Free parking under 30 minutes implemented. Probably need to work on clarity/quality after review

### CODE REVIEW

File: src/main/java/com/parkit/parkingsystem/dao/ParkingSpotDAO.java [hunk 1]
```diff
@@ -38,7 +38,7 @@ public int getNextAvailableSlot(ParkingType parkingType){
     }
 
     public boolean updateParking(ParkingSpot parkingSpot){
-        //update the availability fo that parking slot
+        //update the availability for that parking slot
         Connection con = null;
```
Comment (Asavarkhul) [inline src/main/java/com/parkit/parkingsystem/dao/ParkingSpotDAO.java]: `Connection connection = null;`

File: src/main/java/com/parkit/parkingsystem/dao/TicketDAO.java [hunk 2]
```diff
@@ -19,7 +19,8 @@ public class TicketDAO {
 
     public DataBaseConfig dataBaseConfig = new DataBaseConfig();
 
-    public boolean saveTicket(Ticket ticket){
+    @SuppressWarnings("finally")
+	public boolean saveTicket(Ticket ticket){
```
Comment (Asavarkhul) [inline src/main/java/com/parkit/parkingsystem/dao/TicketDAO.java]: Indentation

Comment (Asavarkhul) [reply src/main/java/com/parkit/parkingsystem/dao/TicketDAO.java]: ){  <- missing space

File: src/main/java/com/parkit/parkingsystem/service/FareCalculatorService.java [hunk 3]
```diff
@@ -10,12 +12,17 @@ public void calculateFare(Ticket ticket){
...
-        int duration = outHour - inHour;
+        long inHour = ticket.getInTime().getTime();
+        long outHour = ticket.getOutTime().getTime();
 
+        //TODO: Some tests are failing here. Need to check if this logic is correct - done
+        double duration = (outHour - inHour)*(1/3.6)*1e-6;
+        System.out.print(duration);
+        if(duration<0.5) {
+        	ticket.setPrice(0.0);
+        }
```
Comment (Asavarkhul) [inline src/main/java/com/parkit/parkingsystem/service/FareCalculatorService.java]: } else {

File: src/main/java/com/parkit/parkingsystem/service/FareCalculatorService.java [hunk 4]
```diff
@@ -10,12 +12,17 @@ public void calculateFare(Ticket ticket){
...
 
-        int inHour = ticket.getInTime().getHours();
-        int outHour = ticket.getOutTime().getHours();
-
-        //TODO: Some tests are failing here. Need to check if this logic is correct
-        int duration = outHour - inHour;
+        long inHour = ticket.getInTime().getTime();
+        long outHour = ticket.getOutTime().getTime();
 
+        //TODO: Some tests are failing here. Need to check if this logic is correct - done
```
Comment (Asavarkhul) [inline src/main/java/com/parkit/parkingsystem/service/FareCalculatorService.java]: Remove todo

File: src/main/java/com/parkit/parkingsystem/service/ParkingService.java [hunk 5]
```diff
@@ -98,12 +98,17 @@ private ParkingType getVehichleType(){
...
+        try{	
             String vehicleRegNumber = getVehichleRegNumber();
             Ticket ticket = ticketDAO.getTicket(vehicleRegNumber);
+            System.out.println(ticket.getId());
+            System.out.println(ticket.getVehicleRegNumber());
             Date outTime = new Date();
             ticket.setOutTime(outTime);
+            
             fareCalculatorService.calculateFare(ticket);
+           // System.out.println("before if");
```
Comment (Asavarkhul) [inline src/main/java/com/parkit/parkingsystem/service/ParkingService.java]: Remove dead code

File: src/test/java/com/parkit/parkingsystem/FareCalculatorServiceTest.java [hunk 6]
```diff
@@ -123,5 +124,36 @@ public void calculateFareCarWithMoreThanADayParkingTime(){
         fareCalculatorService.calculateFare(ticket);
         assertEquals( (24 * Fare.CAR_RATE_PER_HOUR) , ticket.getPrice());
     }
+    
+    @Test
+    public void calculateFareCarUnder30Minutes() {
+
```
Comment (Asavarkhul) [inline src/test/java/com/parkit/parkingsystem/FareCalculatorServiceTest.java:128]: No backspace

File: src/test/java/com/parkit/parkingsystem/integration/ParkingDataBaseIT.java [hunk 7]
```diff
@@ -4,30 +4,50 @@
...
 import static org.mockito.Mockito.when;
 
+import java.text.DateFormat;
+import java.text.DecimalFormat;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+
+
+
```
Comment (Asavarkhul) [inline src/test/java/com/parkit/parkingsystem/integration/ParkingDataBaseIT.java]: useless lines

File: src/test/java/com/parkit/parkingsystem/integration/ParkingDataBaseIT.java [hunk 8]
```diff
@@ -37,29 +57,59 @@ private static void setUp() throws Exception{
 
     @BeforeEach
     private void setUpPerTest() throws Exception {
-        when(inputReaderUtil.readSelection()).thenReturn(1);
+        when(inputReaderUtil.readSelection()).thenReturn(2);
         when(inputReaderUtil.readVehicleRegistrationNumber()).thenReturn("ABCDEF");
-        dataBasePrepareService.clearDataBaseEntries();
+       // dataBasePrepareService.clearDataBaseEntries();
```
Comment (Asavarkhul) [inline src/test/java/com/parkit/parkingsystem/integration/ParkingDataBaseIT.java:55]: Dead code

File: src/test/java/com/parkit/parkingsystem/integration/ParkingDataBaseIT.java [hunk 9]
```diff
@@ -37,29 +57,59 @@ private static void setUp() throws Exception{
...
+        Double realPrice=ticket.getPrice();
+        long inTime=ticket.getInTime().getTime();
+        long duration=outTime.getTime()-inTime;
+        Double exPrice = 1.5*(duration*(1/3.6)*1e-6);
+        double eps = 1e-03;
+      
+        assertEquals(strDateEx,strDateReal);
+        assertEquals(exPrice,realPrice,eps);
+         
         //TODO: check that the fare generated and out time are populated correctly in the database
```
Comment (Asavarkhul) [inline src/test/java/com/parkit/parkingsystem/integration/ParkingDataBaseIT.java]: Remove

**Review by Asavarkhul — CHANGES_REQUESTED** [review]

**Review by Asavarkhul — CHANGES_REQUESTED** [review]
Please

File: src/test/java/com/parkit/parkingsystem/integration/ParkingDataBaseIT.java (hunk 8 above)
Comment (hayfaa14) [reply src/test/java/com/parkit/parkingsystem/integration/ParkingDataBaseIT.java:55]: Can't be deleted right away, will include it in @Before of test for exiting vehicle

**Review by Asavarkhul — APPROVED** [review]

**Pull request merged**